StringArt/StringArt.js

```diff
@@ -0,0 +1,128 @@
+import { Polar } from "../sketchlib/Polar.js";
+
+const MODULUS = 60;
+
+/**
+ * Polynomial ax^3 + bx^2 + cx + d where the coefficients
+ * and values are evaluated mod n
+ */
+class Polynomial {
+  constructor(a, b, c, d) {
+    this.coefficients = [a, b, c, d];
+  }
+
+  compute(x) {
+    const x2 = (x * x) % MODULUS;
+    const x3 = (x2 * x) % MODULUS;
+    const [a, b, c, d] = this.coefficients;
+
+    return (a * x3 + b * x2 + c * x + d) % MODULUS;
+  }
+
+  to_string() {
+    const [a, b, c, d] = this.coefficients;
+    let terms = [];
+
+    if (a != 0) {
+      terms.push(`${a}x^3`);
+    }
+
+    if (b != 0) {
+      terms.push(`${b}x^2`);
+    }
+
+    if (c != 0) {
+      terms.push(`${c}x`);
+    }
+
+    if (d != 0) {
+      terms.push(`${d}`);
+    }
+
+    if (terms.length == 0) {
+      return `0 (mod ${MODULUS})`;
+    }
+    return `${terms.join(" + ")} (mod ${MODULUS})`;
+  }
+}
+
+export const sketch = (p) => {
+  const coefficients = [1, 2, 3, 4];
+  let poly = new Polynomial(...coefficients);
+  let display_sliders = true;
+
+  let polynomial_div = document.getElementById("polynomial");
+
+  p.setup = () => {
+    p.createCanvas(500, 700);
+    polynomial_div.innerText = poly.to_string();
+  };
+
+  p.draw = () => {
```

Comment:
The string art is cool, but likely done before. What can I do to add interest? Some thoughts:

- Animate transitions by lerping points?
- Show powers of the transformation? if so, then `Polynomial` needs a method to compute arrays of the power values (at most `MODULUS`). Each one could be rendered a different shade of the same color
- Extend the lines past the edge of the canvas to make a stellation?

It's worth experimenting with these, but they could potentially be different sketches. This is another reason Polynomial should be its own class.

Reply:
An out-there idea: For each point, compute 4 powers, then draw a bezier curve. Might look interesting, or a mess of spaghetti

Reply:
- plot each orbit a different color?

Reply:
Update:

- the animated transitions is mesmerizing!
- Showing orbits leads to a very busy image... not a huge fan
- The Bezier curve idea occasionally produces a neat pattern, but most of the time it just looks like a mess of spaghetti.

I'm thinking maybe just have the animated transitions. Maybe with a checkbox to toggle it?

Reply:
Looking at this with fresh eyes, I'd say:

- Keep just the animated transitions
- A checkbox wouldn't hurt.